crates/libeq_wld/src/parser/fragments/mesh.rs

```diff
@@ -155,37 +156,39 @@ pub struct MeshFragment {
     /// most illuminated vertices.
     pub vertex_colors: Vec<u32>,
 
-    /// A collection of [MeshFragmentPolygonEntry]s used in this mesh.
-    pub polygons: Vec<MeshFragmentPolygonEntry>,
+    /// A collection of [MeshFragmentFaceEntry]s used in this mesh.
+    pub faces: Vec<MeshFragmentFaceEntry>,
 
     /// The first element of the tuple is the number of vertices in a skeleton piece.
     ///
     /// The second element of the tuple is the index of the piece according to the
     /// [SkeletonTrackSet] fragment. The very first piece (index 0) is usually not referenced here
     /// as it is usually jsut a "stem" starting point for the skeleton. Only those pieces
     /// referenced here in the mesh should actually be rendered. Any other pieces in the skeleton
-    /// contain no vertices or polygons And have other purposes.
-    pub vertex_pieces: Vec<(u16, u16)>,
+    /// contain no vertices or faces And have other purposes.
+    pub skin_assignment_groups: Vec<(u16, u16)>,
 
-    /// The first element of the tuple is the number of polygons that use the same material. All
-    /// polygon entries are sorted by material index so that polygons use the same material are
+    /// The first element of the tuple is the number of faces that use the same material. All
+    /// polygon entries are sorted by material index so that faces use the same material are
     /// grouped together.
     ///
-    /// The second element of the tuple is the index of the material that the polygons use according
+    /// The second element of the tuple is the index of the material that the faces use according
     /// to the [MaterialListFragment] that this fragment references.
-    pub polygon_materials: Vec<(u16, u16)>,
+    pub face_material_groups: Vec<(u16, u16)>,
 
     /// The first element of the tuple is the number of vertices that use the same
-    /// material. Vertex materials, like polygons, are sorted by material index so
+    /// material. Vertex materials, like faces, are sorted by material index so
     /// that vertices that use the same material are together.
     ///
     /// The second element of the tuple is the index of the material that the
     /// vertices use, according to the [MaterialListFragment] fragment that this fragment
     /// references.
-    pub vertex_materials: Vec<(u16, u16)>,
+    pub vertex_material_groups: Vec<(u16, u16)>,
 
-    /// _Unknown_ - A collection of [MeshFragmentData9Entry]s
-    pub data9: Vec<MeshFragmentData9Entry>,
+    /// _Unknown_ - A collection of [MeshFragmentMeshOpEntry]s
+    pub meshops: Vec<MeshFragmentMeshOpEntry>,
+
+    // FIXME: There's some trailing bytes here, between 1-3 in _chr wlds.
```

Comment:
Is this what's causing the new test to fail?
It sounds a lot like padding to make sure this falls on a 4-byte word boundary. Which I _think_ makes sense here. It looks like depending on the `type_field` in the mesh op entries you could end up with a partial word at the end?

Reply:
I'll look into that!  Yes, I made that new test to illustrate this problem.  Is there another fragment with this issue that you resolved?

Reply:
Yep! Check out `user_data` on `RegionFlagFragment`.

The `into_bytes` function on `StringHash` also has to _add_ padding.

Reply:
Just to confirm: I am not actually reading the zeroes when parsing, but I am adding them when serializing.

Reply:
Yep!